sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -0,0 +1,142 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A representation of the model object
+ * <em><b>{@link InputsAdapter InputsAdapter}</b></em>.
+ *
+ * @see LN0Adapter
+ * @see AbstractLNAdapter
+ * @See TInputs
+ * @See TExtRef
+ */
+public class InputsAdapter extends SclElementAdapter<LN0Adapter, TInputs> {
+
+    /**
+     * Constructor
+     *
+     * @param parentAdapter Parent container reference
+     * @param tInputs       Current reference
+     */
+    public InputsAdapter(LN0Adapter parentAdapter, TInputs tInputs) {
+        super(parentAdapter, tInputs);
+    }
+
+    /**
+     * Check if current element is a child of the parent element
+     *
+     * @return true if the currentElem is part of the parentAdapter children
+     */
+    @Override
+    protected boolean amChildElementRef() {
+        return currentElem == parentAdapter.getCurrentElem().getInputs();
+    }
+
+    @Override
+    protected String elementXPath() {
+        return "Inputs";
+    }
+
+    /**
+     * Update iedName of ExtRefs in this "Inputs" element.
+     * @return list of encountered errors
+     */
+    public List<SclReport.ErrorDescription> updateExtRefIedNames() {
+        List<TExtRef> extRefsWithName = getExtRefs().stream().filter(tExtRef -> StringUtils.isNotBlank(tExtRef.getIedName()) && StringUtils.isNotBlank(tExtRef.getDesc())).collect(Collectors.toList());
+        List<TCompasFlow> compasFlows = PrivateService.getCompasPrivates(currentElem, TCompasFlow.class);
+        return extRefsWithName.stream()
+            .map(extRef -> setExtRefName(extRef, compasFlows)).flatMap(Optional::stream).collect(Collectors.toList());
+    }
+
+    /**
+     * List all ExtRefs in this Inputs
+     * @return list of ExtRefs. List is modifiable.
+     */
+    private List<TExtRef> getExtRefs() {
+        if (!currentElem.isSetExtRef()) {
+            return new ArrayList<>();
+        }
+        return currentElem.getExtRef();
+    }
+
+    /**
+     * Find matching CompasFlow private and set ExtRef Name accordingly
+     * @param extRef extRef whose iedName will be updated
+     * @param compasFlows list of all CompasFlow private in this Inputs
+     * @return Error if ExtRef could not be updated
+     */
+    private Optional<SclReport.ErrorDescription> setExtRefName(TExtRef extRef, final List<TCompasFlow> compasFlows) {
+        List<TCompasFlow> matchingCompasFlows = getMatchingCompasFlows(extRef, compasFlows);
+        if (matchingCompasFlows.size() == 1) {
+            TCompasFlow tCompasFlow = matchingCompasFlows.get(0);
+            //TODO: This is not the definitive extRef iedName. The iedName has to be calculated by issue #79(RSR-437)
+            extRef.setIedName("matched " + tCompasFlow.getFlowID());
+            return Optional.empty();
+        } else {
+            return Optional.of(buildErrorWhenNotExactlyOneMatch(extRef.getDesc(), matchingCompasFlows));
```

Comment:
aren't we using too much methods ?  2 methods instead of :  
```java
if (matchingCompasFlows.size() != 1) {
            String message = matchingCompasFlows.isEmpty() ? MESSAGE_EMPTY_COMPAS_FLOW : MESSAGE_TOO_MANY_COMPAS_FLOW;
            return Optional.of(SclReport.ErrorDescription.builder()
                    .xpath(getXPath() + String.format("/ExtRef[%s]", Utils.xpathAttributeFilter("desc", extRef.getDesc())))
                    .message(message)
                    .build());
        } else {
            TCompasFlow tCompasFlow = matchingCompasFlows.get(0);
            //TODO: This is not the definitive extRef iedName. The iedName has to be calculated by issue #79(RSR-437)
            extRef.setIedName("matched " + tCompasFlow.getFlowID());
            return Optional.empty();
        }
```

Reply:
OK